azurelinuxagent/ga/update.py

```diff
@@ -1132,47 +1127,20 @@ def get_agent_cmd(self):
     def get_agent_dir(self):
         return os.path.join(conf.get_lib_dir(), self.name)
 
-    def get_agent_error_file(self):
-        return os.path.join(conf.get_lib_dir(), self.name, AGENT_ERROR_FILE)
-
     def get_agent_manifest_path(self):
         return os.path.join(self.get_agent_dir(), AGENT_MANIFEST_FILE)
 
     def get_agent_pkg_path(self):
         return ".".join((os.path.join(conf.get_lib_dir(), self.name), "zip"))
 
-    def clear_error(self):
-        self.error.clear()
-        self.error.save()
-
     @property
     def is_available(self):
-        return self.is_downloaded and not self.is_blacklisted
-
-    @property
-    def is_blacklisted(self):
-        return self.error is not None and self.error.is_blacklisted
-
-    @property
-    def is_downloaded(self):
-        return self.is_blacklisted or \
-               os.path.isfile(self.get_agent_manifest_path())
-
-    def mark_failure(self, is_fatal=False):
-        try:
-            if not os.path.isdir(self.get_agent_dir()):
-                os.makedirs(self.get_agent_dir())
-            self.error.mark_failure(is_fatal=is_fatal)
-            self.error.save()
-            if self.error.is_blacklisted:
-                logger.warn(u"Agent {0} is permanently blacklisted", self.name)
-        except Exception as e:
-            logger.warn(u"Agent {0} failed recording error state: {1}", self.name, ustr(e))
+        return os.path.isfile(self.get_agent_manifest_path())
```

Comment:
what does "available" mean? how is it different to "downloaded"?

Reply:
earlier available meant downloaded and not blacklisted and downloaded meant the availability of the manifest file inside the agent directory. Since blacklisted is not relevant anymore, removed `is_downloaded` and using `is_available` everywhere.